scripts/run-related-test.mjs

```diff
@@ -0,0 +1,46 @@
+/**
+ * This script finds all `.tests` files under the paths and reads lines from these files.
+ * The script is useful to find all related test cases for a given code change.
+ * Usage: `node scripts/run-related-test.mjs <path1> <path2> ...`
+ */
+
+import { promisify } from 'node:util'
+import { exec as execOrg } from 'node:child_process'
+import { readFile, constants, access } from 'node:fs/promises'
+import { dirname, join } from 'node:path'
+
+const exec = promisify(execOrg)
+
+/**
+ * Get all changed files from git under the packages directory
+ * @returns {Promise<string[]>} - List of changed files
+ */
+async function getChangedFilesFromPackages(baseBranch = 'origin/canary') {
+  const { stdout } = await exec(`git diff --name-only ${baseBranch}`)
+  return stdout
+    .trim()
+    .split('\n')
+    .filter((line) => line.startsWith('packages/'))
+}
+
+export async function getRelatedTests(args = []) {
+  const paths = args.length ? args : await getChangedFilesFromPackages()
+  const relatedTestFile = '.tests'
+  const tests = []
+  for (const path of paths) {
+    const testFile = join(dirname(path), relatedTestFile)
+    const hasRelatedTests = await access(testFile, constants.F_OK)
```

Comment:
Using access() and then readFile() can cause race conditions

> Using fsPromises.access() to check for the accessibility of a file before calling fsPromises.open() is not recommended. Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file is not accessible.

https://nodejs.org/api/fs.html#fspromisesaccesspath-mode